### PR TITLE
Rooms

### LINKED ISSUE #56: Add Rooms
-Server-side
  - Organize Rooms - each room has it's own game instance
  - be able to return room list
  - be able to join a specific room
  - be able to create and join a new room
  - after joined, the player should pass it's `id` and the appropriate room will be updated / have info returned

- Client-side
  - fetch and display room in lobby
  - be able to join a room
  - once joined, be able to select `generator` role if it's open
  - play as normal after that

### PR DESCRIPTION
**Pull request by GreyMatteOr**
## Description

Refactored to work with new `Room` architecture. Added a Lobby to join Rooms / Create Rooms.
Also disabled the generator button when a room has a generator.

## Context

closes #56 
[Corresponding API PR](https://github.com/GreyMatteOr/bangwords-api/pull/11)

## Affected areas of application

In the new Lobby Class

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [] In Browser with Other instance of client and local server running

## Questions or Notes (if applicable):

- Still need to add a leave room button
- need to write and update tests

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import React, { Component } from 'react';
-import { Switch, Route } from 'react-router-dom';
+import { Route } from 'react-router-dom';
 import  { Homepage }  from '../src/Homepage/Homepage.js';
 import  { WordSelector }  from '../src/WordSelector/WordSelector.js';
 import  { Gamepage }  from '../src/Gamepage/Gamepage.js';
+import  { Lobby }  from '../src/Lobby/Lobby.js';
```
Comment (holladayian) [inline src/App.js:6]: ooo lala lobby

File: src/WordSelector/WordSelector.js [hunk 2]
```diff
@@ -38,7 +38,7 @@ export class WordSelector extends Component {
               placeholder="Submit"
               onClick={(e) => {
                 this.makeWordToGuess(e)
-                }}
+              }}
```
Comment (holladayian) [inline src/WordSelector/WordSelector.js:41]: really cutting down on data loss here. We have to do these things or big data wins

**Review by holladayian — APPROVED** [review]
Looks good! Nice clean work. Has my approval!

File: src/Lobby/Lobby.css [hunk 3]
```diff
@@ -0,0 +1,34 @@
+.lobby {
```
Comment (Chriscastanuela) [inline src/Lobby/Lobby.css:1]: this is very well designed

**Review by Chriscastanuela — APPROVED** [review]

**Pull request merged**